.github/workflows/docker_publish.yml

```diff
@@ -1,40 +1,209 @@
+  pull_request:
+  push:
+    branches:
+      - master
   release:
-    types: [ released ]
+    types:
+      - released
+
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  push_server_to_registry:
-    name: Push Docker server image to Docker Hub
+  # This job exposes image_tag and push_enabled variables for other jobs.
+  # Other jobs depends on this. see needs: meta in other jobs.
+  meta:
+    runs-on: ubuntu-latest
+    outputs:
+      image_tag: ${{ steps.determine-image-tag.outputs.image_tag }}
+      push_enabled: ${{ steps.determine-push.outputs.enabled }}
+    steps:
+      # this step determines the image tag based on the event that triggered the workflow.
+      # it sets the image_tag output variable to either "master" or the release version.
+      - name: Determine image tag
+        id: determine-image-tag
+        run: |
+          if [[ "${{ github.event_name }}" == "push" ]]; then
+            echo "image_tag=master" >> $GITHUB_OUTPUT
+          elif [[ "${{ github.event_name }}" == "release" ]]; then
+            echo "image_tag=${{ github.event.release.tag_name }}" >> $GITHUB_OUTPUT
+          else
+            echo "image_tag=test" >> $GITHUB_OUTPUT
+          fi
+
+      # This step determines if push is enabled based on the event that triggered the workflow. Push is disabled on pull requests.
+      - name: Determine if push is enabled
+        id: determine-push
+        run: |
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            echo "enabled=false" >> $GITHUB_OUTPUT
+          else
+            echo "enabled=true" >> $GITHUB_OUTPUT
+          fi
+
+  # TODO: Delete below job below once image tags are validated after merging
+  check-meta:
+    needs: meta
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check meta
+        run: |
+          echo "Image tag: ${{ needs.meta.outputs.image_tag }}"
+          echo "Push enabled: ${{ needs.meta.outputs.push_enabled }}"
+      - name: This step should run if push is enabled
+        if: ${{ needs.meta.outputs.push_enabled == 'true' }}
+        run: |
+          echo "Push enabled: ${{ needs.meta.outputs.push_enabled }}"
+      - name: This step should run if push is disabled
+        if: ${{ needs.meta.outputs.push_enabled == 'false' }}
+        run: |
+          echo "Push enabled: ${{ needs.meta.outputs.push_enabled }}"
+
+  push_server_image:
+    name: Push server image to Docker Hub
+    needs: meta
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Login to Docker Hub
+        if: ${{ needs.meta.outputs.push_enabled == 'true' }}
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.CADENCE_SERVER_DOCKERHUB_USERNAME }}
+          password: ${{ secrets.CADENCE_SERVER_DOCKERHUB_TOKEN }}
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
```

Comment:
Nice, I must have missed this step when I tried to publish multi-plat images, I was getting buildx not supported errors